src/Sylius/Bundle/CoreBundle/Resources/config/app/state_machine.yml

```diff
@@ -37,14 +37,14 @@ winzou_state_machine:
                 to: completed
         callbacks:
             after:
+                sylius_update_exchange_rate:
+                    on: ["select_payment"]
+                    do: ["@sylius.updater.order.exchange_rate_and_currency", "update"]
+                    args: ["object"]
```

Comment:
Until #5847 gets merged unfortunately it is necessary, otherwise after the `select_payment` transition the cart currency code and exchange rate reset to the default values.